### PR TITLE
Move tabs at top of window and update icons

### PR DESCRIPTION
**Pull request by 4ian**
* Move the tabs in the titlebar. Handle window controls both on desktop and as a PWA.
* Update all icons of the toolbars for modern icons. 
* Add a menu that will replace the application menu on desktop (except on macOS) and which will work on the web-app
  * Remove the ugly save/save as in the web-app that were in the project manager.
* Adapt dialogs and the project manager to display a blank space for the titlebar/window controls if needed (see some screenshots).
* Replace all the three dots icons (not visible on the screenshots here).

TODO:
- [x] Update Project manager icons
- [x] fix storybook
- [x] Fix pause/play button in debugger

| macOS desktop  | Windows desktop |
| ------------- | ------------- |
| image  | image  |

| Chrome PWA macOS Classic  | Chrome PWA macOS overlay |
| ------------- | ------------- |
| image  | image  |

Chrome PWA Windows:
image

Safari macOS:
image

Mobile:

| Chrome PWA Android  | Chrome PWA Android (2) |
| ------------- | ------------- |
| Screenshot_20221206-182946 | Screenshot_20221206-182924 |

Debugger:
image

Dialogs on small screens:
| Desktop (macOS here)  | Chrome PWA |
| ------------- | ------------- |
| image | image |

Project manager:
| Desktop (Windows here)  | Chrome PWA |
| ------------- | ------------- |
| image | image|

### CODE REVIEW

**Comment by tristanbob** [comment]
How is the Network Preview accessed?  (I think it should be a split button on Preview, instead of the Debugger button)

**Comment by 4ian** [comment]
Yep very good remark, I've now moved the old "debug" button to the preview button context menu: 
image

There is not "split" arrow displayed for now, we might add it later but for now it's still discoverable thanks to the tooltip, and this is pretty advanced stuff anyway.

File: newIDE/app/src/MainFrame/ElectronMainMenu.js [hunk 1]
```diff
@@ -82,303 +24,69 @@ const useIPCEventListener = (ipcEvent: MainMenuEvent, func) => {
...
+            shortcutMap,
+            isApplicationTopLevelMenu: true,
+          })
+        );
+      }
     },
     [i18n, language, project, recentProjectFiles, shortcutMap]
   );
 
+  // TODO: move this? And ensure stability.
```
Comment (ClementPasteau) [inline newIDE/app/src/MainFrame/ElectronMainMenu.js:88]: not sure about what this comment means.
If it's important to be done later, I'd make the message clearer.

File: newIDE/app/src/MainFrame/index.js [hunk 2]
```diff
@@ -2663,33 +2654,42 @@ const MainFrame = (props: Props) => {
...
+    onOpenAbout: () => openAboutDialog(true),
+    onOpenPreferences: () => openPreferencesDialog(true),
+    onOpenLanguage: () => openLanguageDialog(true),
+    onOpenProfile: () => openProfileDialogWithTab('profile'),
+    onOpenGamesDashboard: () => openProfileDialogWithTab('games-dashboard'),
+    setElectronUpdateStatus: setElectronUpdateStatus,
+  };
+
   return (
     <div className="main-frame">
```
Comment (ClementPasteau) [inline newIDE/app/src/MainFrame/index.js:2687]: what's this className for?
If it's important, maybe a small comment explaining why I shouldn't remove it

File: newIDE/app/src/UI/IconButton.js [hunk 3]
```diff
@@ -54,8 +54,7 @@ type Props = {|
   acceleratorString?: string,
   'aria-label'?: string,
 
-  /** To be used only when button should have a constant color, independently from the theme. */
-  useCurrentColor?: boolean,
```
Comment (ClementPasteau) [inline newIDE/app/src/UI/IconButton.js:58]: this was never used?

File: newIDE/app/src/UI/RaisedButton.js [hunk 4]
```diff
@@ -3,6 +3,8 @@ import * as React from 'react';
 import Button from '@material-ui/core/Button';
 import { type ButtonInterface } from './Button';
 import { Spacer } from './Grid';
+import { Tooltip } from '@material-ui/core';
```
Comment (ClementPasteau) [inline newIDE/app/src/UI/RaisedButton.js:6]: prefer `import Tooltip from '@material-ui/core/Tooltip'`;

File: newIDE/app/src/UI/FlatButton.js [hunk 5]
```diff
@@ -3,6 +3,8 @@ import * as React from 'react';
 import Button from '@material-ui/core/Button';
 import { type ButtonInterface } from './Button';
 import { Spacer } from './Grid';
+import { Tooltip } from '@material-ui/core';
```
Comment (ClementPasteau) [inline newIDE/app/src/UI/FlatButton.js:6]: prefer `import Tooltip from '@material-ui/core/Tooltip'`;

File: newIDE/app/src/UI/Theme/DefaultDarkTheme/theme.json [hunk 6]
```diff
@@ -394,12 +414,6 @@
       }
     },
     "home": {
-      "header": {
```
Comment (ClementPasteau) [inline newIDE/app/src/UI/Theme/DefaultDarkTheme/theme.json:397]: I believe you haven't removed it from all theme.json files (ex: BlueDarkTheme)

File: newIDE/app/src/Utils/Window.js [hunk 7]
```diff
@@ -42,16 +84,19 @@ export default class Window {
...
-
     if (currentTitleBarColor === newColor) {
       // Avoid potentially expensive DOM query/modification if no changes needed.
       return;
     }
 
+    if (ipcRenderer) {
+      ipcRenderer.invoke('titlebar-set-overlay-options', {
+        color: newColor,
+        symbolColor: '#ffffff',
```
Comment (ClementPasteau) [inline newIDE/app/src/Utils/Window.js:95]: what does the symbol represent?
Is there a risk it's visible if it's always white?

File: newIDE/app/src/stories/index.js [hunk 8]
```diff
@@ -2903,99 +2903,6 @@ storiesOf('BrowserPreviewErrorDialog', module)
     />
   ));
 
-storiesOf('ProjectManager', module)
```
Comment (ClementPasteau) [inline newIDE/app/src/stories/index.js:2906]: 🤟

**Review by ClementPasteau — APPROVED** [review]
neat!

File: newIDE/app/src/MainFrame/ElectronMainMenu.js (hunk 1 above)
Comment (4ian) [reply newIDE/app/src/MainFrame/ElectronMainMenu.js:88]: Removed the comment. We might at some point move the command palette commands to open recent files, but as this make less sense for now in the web-app, not a big deal.

File: newIDE/app/src/MainFrame/index.js (hunk 2 above)
Comment (4ian) [reply newIDE/app/src/MainFrame/index.js:2687]: Adding a comment

File: newIDE/app/src/UI/RaisedButton.js (hunk 4 above)
Comment (4ian) [reply newIDE/app/src/UI/RaisedButton.js:6]: Good catch thanks

File: newIDE/app/src/Utils/Window.js (hunk 7 above)
Comment (4ian) [reply newIDE/app/src/Utils/Window.js:95]: Good catch. I've added a logic to display the controls in black if the background is dark.

**Comment by LuniMoon** [comment]
Design micro changes:

- [ ] Move scene editor (currently on the right at the end of the bar) in front of objects
- [ ] Allowing to explore the options inside the burger menu with hovers (actual interaction demands to click on each section to see what's inside). Which will fix the bad UX of having to click 3 times outside the menu to close it.
- [ ] Add an asterisk to the end of the name on the tab to indicate "unsaved changes"
- [ ] Tabs are responsive (like Chrome's example)
Screenshot 2022-12-08 at 11 27 39

- [ ] Adding the "we've moved these actions to the menu" for Save, Save As, etc... if possible. And/or showing them as unable options. Also, showing the commands for those actions would be fantastic (that way we don't just "take", but give something in return).
- [ ] I'm for putting the debugger under preview (cause it is an advanced feature), but the button doesn't tell the users that there are more options hidden (only those who have discovered them know!). I'd use this button instead:
Screenshot 2022-12-08 at 10 51 15

- [ ] Why is the search extension component different than the search component reworked by Clem ? If possible, replace it by the standardised component.
Screenshot 2022-12-08 at 11 46 00

I'm making a mobile version for the bar.

**Comment by jack-ii** [comment]
> For this, I recommend you check the command palette and the associated customisable shortcut.

In my opinion, hot-reload is the advanced feature and a new preview is the most needed normal case.

I might use the shortcut in the future. But, well, it was possible for years to do it with the mouse and a preferences switch has no disadvantages.. Okay was worth a try.

**Comment by LuniMoon** [comment]
I've finished the mobile version for the bars (you can find them on the "Layout Improvement" Figma file
More design optims (to go the extra mile):

- [ ] For the panels: replace "Open settings" by "Open Scene Properties" (that's the icon for Scene manager)
- [ ] Implement the "on/off" (clicked) button interactions for the panels 
Screenshot 2022-12-09 at 12 12 35

**Comment by jack-ii** [comment]
I think the "Publish" button background should be the same as the "Preview" next to it. This button is only used once after the game is finished (in my opinion it shouldn't even be in this menu), but has more focus than anything else in GD. Much more than active tabs, etc.

**Pull request merged**